integration-libs/s4-service/order/occ/adapters/occ-reschedule-service-order.adapter.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import { RescheduleServiceOrderAdapter } from '../../core/connector/reschedule-service-order.adapter';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import {
+  LoggerService,
+  normalizeHttpError,
+  OccEndpointsService,
+} from '@spartacus/core';
+import { ServiceDetails } from '@spartacus/s4-service/root';
+import { catchError, Observable } from 'rxjs';
+
+const CONTENT_TYPE_JSON_HEADER = { 'Content-Type': 'application/json' };
+
+@Injectable()
+export class OccRescheduleServiceOrderAdapter
+  implements RescheduleServiceOrderAdapter
+{
+  protected http = inject(HttpClient);
+  protected logger = inject(LoggerService);
+  protected occEndpoints = inject(OccEndpointsService);
+
+  rescheduleServiceOrder(
+    userId: string,
+    code: string,
+    scheduledAt: ServiceDetails
+  ): Observable<unknown> {
+    const url = this.occEndpoints.buildUrl('rescheduleService', {
+      urlParams: { userId, code },
+    });
+    const headers = new HttpHeaders({
+      ...CONTENT_TYPE_JSON_HEADER,
```

Comment:
What this " ... suggests"

Reply:
`...` is a spread operator, duplicating the contents of `CONTENT_TYPE_JSON_HEADER` rather than directly using the identifier.

Reply:
My question is more about whether you need to duplicate it when instantiating. Is it necessary to use it in this context?

Reply:
In this context, we could directly use the value instead of assigning it to a constant first and then duplicating the values where needed.
I copied this part from another place in the source code assuming it to be some best practise or convention followed in Spartacus.

Reply:
Updated.